### ISSUE TITLE
Unable to build CostFed from Maven

### ISSUE THREAD

**Issue by natancox** [dice-group/CostFed#2]
Some problems prevent CostFed to be build. It is somewhat difficult to offer suggestions since it is not clear to me what your plans for a build setup are.

Here are some issues I am having at the moment

- In root pom.xml you refer to `costfed-web` where the folder currently is named 'co**n**stfed-web'
- In pom of fedx artifacts for commons lang3 (org.apache.commons:commons-lang3) and rdf4j.sail.nativerdf (org.eclipse.rdf4j:rdf4j.sail.nativerdf) are not configured. Possibly some other artifacts are missing too.
- Encoding is not set in Maven which makes the build platform dependent: different interpretation of your source code for Windows and Linux!

**Comment by apotocki** [comment]
> In root pom.xml you refer to costfed-web where the folder currently is named 'constfed-web'

yes, constfed-web is a typo. But, in fact, root pom.xml is used as a common dependencies holder for all its submodules. So that typo didn't cause any problem. Any way I fixed it.

> In pom of fedx artifacts for ....

I don't understand why we need to configure "commons lang3" and rdf4j.sail.nativerdf. These libs are dependencies of rdf4j-runtime artifact that is configured in our pom.

> Encoding is not set in Maven which makes the build platform dependent: different interpretation of your source code for Windows and Linux!

I can imagine only one case when a java source code would have a different interpretation depending on the encoding. This is the usage of not ascii symbols in string literals and identifiers. But we code only in ascii. So I think it's a minor bug that cannot prevent from building the code for those both platforms.

**Comment by natancox** [comment]
I attached my build output for only the "fedx" component. Cfr  [fedx-build.log](https://github.com/dice-group/CostFed/files/1494840/fedx-build.log). It clearly shows the errors.

Please note that "runtime" means the component is **not required** for compilation. But since the code uses some of those classes it should be marked as compile. See for example this explanation on [stackoverflow](https://stackoverflow.com/a/19541909/328808).

It might be that you do not use maven but some other tool  to build.

**Comment by apotocki** [comment]
Thank you for remarks. We use eclipse to develop these projects and haven't tried to build them from the command line. This IDE seems to consider all dependencies as 'compile' ones no matter how they are declared. So I'll try to resolve this dependency issues shortly.
